### ISSUE TITLE
[W10] Transparent instead of Blurred Window

### ISSUE THREAD

**Issue by blryface** [PSJahn/blurredwindow#4]
According to the Modrinth description, it should work on either windows 10 or 11. I've installed the mod and it doesn't seem to be the case.

![Blurred Window mod on Windows 10](https://github.com/PSJahn/blurredwindow/assets/124838505/0ab5a85f-4bbd-4618-8075-2dc6baada671)

I'm using the [Additive](https://modrinth.com/modpack/additive) modpack, but I'll test this in a "vanilla" install to see if it works.

**Comment by blryface** [comment]
It does not work on a fresh install of Fabric 1.20.6, with only Fabric and the mod loaded either.
![Blurred Window mod on Windows 10, on a "vanilla" client](https://github.com/PSJahn/blurredwindow/assets/124838505/c9b67924-a652-44b9-ac6f-b449448560d2)

**Comment by PSJahn** [comment]
Could you tell me your Windows build version?

**Comment by blryface** [comment]
Sure! Here's Winver and neofetch.

![winver](https://github.com/PSJahn/blurredwindow/assets/124838505/3171be04-436a-4df7-af88-7ed98a9d660b)
![neofetch](https://github.com/PSJahn/blurredwindow/assets/124838505/bc1b207b-4ef8-4629-a966-9adf6bd70a4d)

**Comment by blryface** [comment]
Maybe windows update is just borked, so i'll try installing this supposedly optional update
![Windows Update](https://github.com/PSJahn/blurredwindow/assets/124838505/f97811e1-0f88-40fd-8307-9c1a0c5c6a1c)

**Comment by blryface** [comment]
I forgot to test again that day, but testing again today shows it hasn't fixed my issue.
![Test post-update](https://github.com/PSJahn/blurredwindow/assets/124838505/f70e2747-b66a-4b49-9d54-bd74109696f3)
Before clicking the window the title bar also has some weird glitch, but I think this is windows' fault

**Comment by EnderKill98** [comment]
This is an odd problem. I think there are a few troubleshooting steps that can get us closer to what is the issue here.

## 1. Check if Windows "Fluent Acrylic" works in general.

Your version seems to be the most recent Windows version. That version being officially released in Q2 of 2022 isn't telling much though. Windows fluent design stuff is pretty new and I'd assume it wasn't full ready then. So it might have only shipped in updates.

Can you try to check if any other app can use the Acrylic Background? This would be a good indicator, whether your Windows installation in general has the ability or not. For me, the new Windows Terminal comes to mind. But feel free to test other apps, that should have the background.

## 2. Does the mod output any error to the console, when trying to enable Acrylic Background?

I'm not familiar with the new Minecraft Launcher. But there should probably be some way to access the game logs through it. Worst case you need to go to your `.minecraft` folder in `%APPDATA%` and check the contents of `logs\latest.log` (text file).

There might be some error related to the mod. If you could find any, this would be a huge help.

## 3. Is this a regression of checking for the platform?

This is a far fetch, but I recently contributed adding blur to Linux (KDE) as well. With that I made the Windows calls only happen, if GLFW determines the Platform to be Win32. Afaik this is the only platform corresponding to windows. But I might be wrong.

Can you try out, if this version (before these changes), behaves any differently?
https://modrinth.com/mod/blurred-window/version/1.0.1

## 4. Are any other mods causing this perhaps?

I doubt this will be the case, since your instance seems pretty lightly modded, but this could also maybe the culprit.

It might be possible, that some other mod is preventing this mod from running the code, which instructs Windows to use the acrylic background (but doesn't interfere with the transparency otherwise)

Can you try out, if the mod works, when all/most other mods are disabled?

**Comment by blryface** [comment]
1. Yes, other native apps use similar functions
2. I don't see any errors from anything, I'm also using the modrinth app, so the instructions are kind of irrelevant in this case.
3. That.. shouldn't cause any issues? It probably doesn't just not see I'm running Win32
4. I've showed in 2 of the screenshots a completely vanilla game, apart from the fabric loader and Blurred Window

**Comment by EnderKill98** [comment]
> 1. Yes, other native apps use similar functions

That indicates that it should indeed work then. Then it's probably some weird case of, it being implemented the wrong way. I recently saw some blog posts over it. There are a lot of ways to get the "Acrylic" effect. It can be super detailed or not, and there are multiple other versions such "Mica" that look similar to acrylic, but are something else entirely.

> 2. I don't see any errors from anything, I'm also using the modrinth app, so the instructions are kind of irrelevant in this case.

I wouldn't have known, how to get the logs on the official launcher anyway. I think I launched it once on linux and that was it. I used to love the old Java-based launcher. But nowadays I stick to MultiMC or it's derivatives.

> 3. That.. shouldn't cause any issues? It probably doesn't just not see I'm running Win32

It is a possibility. Unlikely, but the window being transparent it done always. Reuqesting that transparent background to become acrylic is a different check where there was no platform check, but now it is.

> 4. I've showed in 2 of the screenshots a completely vanilla game, apart from the fabric loader and Blurred Window

Ah nice. It's hard to tell from these if there are any mods to me. That's why I asked.


---

So in summary, it should work. I don't know then why this wouldn't be the case. I'm not deep into windows myself. I mainly use it for work occasionally and haven't gamed on it in almost 10 years.

But thanks for providing the information. I'd assume there needs to be more testing to see if this is actually on every Windows 10 Installation, or just some subset and why.

**Comment by blryface** [comment]
I'm gonna check on my laptop soon, if it works i'll tell yall

**Comment by blryface** [comment]
It doesn't work there either

**Comment by blryface** [comment]
Hey it's been a month. Do y'all have any updates on this?

**Comment by PSJahn** [comment]
nope absolutely clueless

**Comment by blryface** [comment]
huh. I will try again to see if it was an issue specific to when I was on that windows install

**Comment by PSJahn** [comment]
I mean it's definitely specific to you as many people are using the mod fine with windows 10 lol

**Comment by liquidsquid1** [comment]
can replicate this issue on my host (all performance related stuff is off - windows is using its best 'appearance')
no errors to the console, no other mods, other apps can use acrylic, the mod just straight up doesn't work :(

**Comment by blryface** [comment]
> I mean it's definitely specific to you as many people are using the mod fine with windows 10 lol

Well apparently you're clearly wrong

**Comment by PSJahn** [comment]
> > I mean it's definitely specific to you as many people are using the mod fine with windows 10 lol
> 
> Well apparently you're clearly wrong

Well yes, but actually no.
This is very rare as there have been hundreds of people using the mods without any issues, and I've personally tested it with about 20 people varying from various linux distros to windows 10 to windows 11.

There is no apparent reason for the mod not to work, as It's simply calling some functions of the Windows Desktop Manager DLL, and if there's no errors, the issues isn't with the mod calling the functions, but rather some other windows shenanigans, which I cannot debug.
So for the time being, I have absolutely no way of debugging these issues, and therefore no way of fixing them either.

Feel free to open a pull request if you have any better ideas on what the issue could be caused by, and are willing to fix it.
